crates/esplora/src/blocking_ext.rs

```diff
@@ -7,12 +7,54 @@ use bdk_core::{
     BlockId, CheckPoint, ConfirmationBlockTime, Indexed, TxUpdate,
 };
 use esplora_client::{OutputStatus, Tx};
+use std::any::Any;
+use std::fmt;
 use std::thread::JoinHandle;
 
 use crate::{insert_anchor_or_seen_at_from_status, insert_prevouts};
 
-/// [`esplora_client::Error`]
-pub type Error = Box<esplora_client::Error>;
+#[derive(Debug)]
+pub enum Error {
+    Client(esplora_client::Error),
+    ThreadPanic(Option<String>),
```

Comment:
Instead of having this, can we just ensure that the thread does not panic?

Reply:
You're right. I took your advice and removed `ThreadPanic`